iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -371,7 +371,7 @@ cxx::expected<IpcChannelError> UnixDomainSocket::initalizeSocket(const IpcChanne
         auto connectCall =
             posixCall(connect)(m_sockfd, reinterpret_cast<struct sockaddr*>(&m_sockAddr), sizeof(m_sockAddr))
                 .failureReturnValue(ERROR_CODE)
-                .evaluateWithIgnoredErrnos(ENOENT);
+                .evaluateWithIgnoredErrnos(ENOENT, ECONNREFUSED);
```

Comment:
I'm unsure if I like the idea. Then the decision is not binary anymore and we have three categories: errors, silent errors and ignored errors. In this case it would just mean the check for `ECONNREFUSED` is moved from happy path to the sad path, right?

Reply:
@elBoberido I think I am slightly against it. But I would support that we get rid of `evalluateWithIgnoredErrnos` and replace it with `evaluateWithSilentErrnos` I feel that it would make much more sense. Then we could get rid of the errnum in the success case since here it would be always `errno = 0`.
Yeah I like it!